### PR TITLE
Form Rule mutations

### PR DESCRIPTION
**Pull request by gigxz**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Mutations for create/update form rules. This is a V1, still needs more features before we enable it.

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/graphql/types/forms/enums/data_collected_about.rb [hunk 1]
```diff
@@ -15,8 +15,6 @@ class Forms::Enums::DataCollectedAbout < Types::BaseEnum
       'HOH_AND_ADULTS',
       'HOH',
       'ALL_VETERANS',
-      'ALL_CLIENTS_RECEIVING_SSVF_SERVICES',
-      'ALL_CLIENTS_RECEIVING_SSVF_FINANCIAL_ASSISTANCE',
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/forms/enums/data_collected_about.rb:19]: Removing these as they are unused

**Review by eanders — APPROVED** [review]
Can't say I'm following what's going on, but it looks good to me.  Thanks for the fix on Prior Living Situation language.

**Comment by gigxz** [comment]
Oops I didn't mean to include that PLS change, but oh well. I will add the rest of the wording changes separately.

The mutations in this PR support an interface where you can create/edit the Form Instances. Those are the things that specify "only collect CLS for HoH in CE projects" or "use this special form for Project editing in this Organization" etc. The UI is in progress~

File: drivers/hmis/app/graphql/mutations/create_form_rule.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      raise 'not allowed' unless current_user.can_configure_data_collection?
+
+      definition = Hmis::Form::Definition.find(definition_id)
+
+      instance = Hmis::Form::Instance.new(definition_identifier: definition.identifier)
+      instance.assign_attributes(input.to_attributes)
+
+      if instance.valid?
+        instance.save!
+        { form_rule: instance }
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/mutations/create_form_rule.rb:24]: Micro-optimization here, rails save() calls validate() and returns true/false if validation fails. So you can say:
```ruby
      if instance.save
          { form_rule: instance }
```

File: drivers/hmis/app/graphql/mutations/create_form_rule.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class CreateFormRule < BaseMutation
```
Comment (ttoomey) [inline drivers/hmis/app/graphql/mutations/create_form_rule.rb:8]: Maybe use CleanBaseMutation to avoid autogenerated CreateFormRuleInput type?

**Review by ttoomey — COMMENTED** [review]
Looks good!

**Pull request merged**